### DISCUSSION TITLE
Binding Issue

### DISCUSSION THREAD

**Discussion by End3r6** [AvaloniaUI/Avalonia#8475] [Q&A] [answered]
K, I am not sure if this is a bug (probably not) but binging refuses to work for me. I have looked all over the place for a solution and haven't found one that works yet. Any help would be greatly appreciated.

Essentially, I have a list and want to send that data over to a List Box. This should be simple but I can't get this working for the life of me.

This is my axaml

```
<Design.DataContext>
    <vm:SongsListViewModel/>
  </Design.DataContext>

  <StackPanel>
    <TextBlock Text="Songs" Margin="10" FontSize="18"/>
    <ListBox x:Name="SongsDisplay" Background="#383638" Items="{Binding SongsDisplayList}">
        <ListBox.ItemTemplate>
            <DataTemplate>
                <Border CornerRadius="4" Padding="4">
                    <TextBlock Text="Cheese"/>
                </Border>
            </DataTemplate>
        </ListBox.ItemTemplate>
    </ListBox>
  </StackPanel>

```
And this is my view model

```
private AvaloniaList<Song> songsDisplayList = new AvaloniaList<Song>();
        
public AvaloniaList<Song> SongsDisplayList
{
      get => songsDisplayList;
      set => this.RaiseAndSetIfChanged(ref songsDisplayList, value);
}

public SongsListViewModel()
{
      SongsDisplayList = SongManager.songs;
}

```
Again, any help would be greatly appreciated and I am sorry if this has already been discussed and I just didn't find it.

**Comment by maxkatz6** [comment, marked as answer]
From this code, it should work just fine.
You set `Design.DataContext` there, which is fine for the XAML designer. 
But do you set data context on the window normally? I.e., on window creation - `new MainWindow { DataContext = new SongsListViewModel() }`.

If you want to set it from xaml, you also can do it:
```
<Window.DataContext>
    <vm:SongsListViewModel/>
  </Window.DataContext>
```
Note, that I don't use `Design.DataContext` here. `Design` properties are only applied when app is run from the designer.

Same applies for user controls too.

**Reply by End3r6** [reply]
Hey, so I tried the above c# method for setting the design context in the new main window declaration, and it didn't work. I am using a user control for the list and my machine is a mac, would that affect anything. Also thanks for the response. :)

**Reply by End3r6** [reply]
would it be better to keep everything that is on one window In the main window axaml file?

**Reply by FoggyFinder** [reply]
Can you provide [MRE](https://software.codidact.com/posts/285810/285811#answer-285811)?. Best if you create github repo with test project. 

Also, check [ToDo List App](https://docs.avaloniaui.net/tutorials/todo-list-app) from the docs to get better understanding how bindings work.

**Reply by End3r6** [reply]
I see, I was originally using the todo list app. Here is my repo stuff. [Xaml](https://github.com/End3r6/AudioVine/blob/master/Views/MainWindow.axaml)
[ViewModel](https://github.com/End3r6/AudioVine/blob/master/ViewModels/MainWindowViewModel.cs)

**Reply by maxkatz6** [reply]
`SongManager.Create()` is never executed in your code.

**Reply by maxkatz6** [reply]
Also, if you want to read these files from the file system, and not assembly assets, change these files' properties to something like this:
![image](https://user-images.githubusercontent.com/3163374/178172618-40715852-f3bb-490e-bd1e-93fc3ab0e228.png)

It should be Content and Copy if never

**Reply by End3r6** [reply]
Thank you so much, this worked. I now feel very dumb lol. Quick question, I am on vscode, how would I set the properties to content and copy if never?

**Reply by End3r6** [reply]
Never mind, found it :)

**Comment by End3r6** [comment]
[MRE:](https://github.com/End3r6/AudioVine)
